app/main.py

```diff
@@ -1,6 +1,19 @@
-from typing import Callable
+from typing import Callable, Any
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cached_results = {}
+
+    @wraps(func)
+    def wrapper(*args) -> Any:
+        for arguments, result in cached_results.items():
```

Comment:
here you can just check if `key` in `cached_results`, so no need to use `items()` method